src/lib/homestore_backend/index_kv.cpp

```diff
@@ -70,7 +71,59 @@ HSHomeObject::get_from_index_table(shared< BlobIndexTable > index_table, shard_i
         return folly::makeUnexpected(BlobError::UNKNOWN_BLOB);
     }
 
-    return index_value.pbas();
+    const auto& pbas = index_value.pbas();
+
+    // blob get API
+    if (state == BlobState::ALIVE) {
+        if (pbas == tombstone_pbas) { return folly::makeUnexpected(BlobError::INDEX_ERROR); }
+        return pbas;
+    }
+
+    // get tombstone, only used by test for now, can be used by GC thread if needed
+    if (state == BlobState::TOMBSTONE) {
+        if (!(pbas == tombstone_pbas)) { return folly::makeUnexpected(BlobError::INDEX_ERROR); }
+    }
+
+    // used for GC thread to range scan index table
+    return pbas;
+}
+
+// this is used for GC thread to delete tombstone blobid from btree
+BlobManager::NullResult HSHomeObject::delete_from_index_table(shared< BlobIndexTable > index_table, shard_id_t shard_id,
```

Comment:
This is not needed, as for GC, it will call index range-update and in the filter callback, it will return remove method for the tombstoned values and btree will remove the tombstoned keys/values atomically.